src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java

```diff
@@ -53,7 +54,12 @@
     );
 
     private static final Set<String> stringOperators = Sets.newHashSet(
```

Comment:
Seems like we starting to have considerable duplication here with our grammar file.
This is not a request for this PR, but if that information is somehow . For example those sets doesn't give us anything useful
```
    private static final Set<String> stringOperators = Sets.newHashSet(
            "split", "concat_ws", "substring", "trim", "lower", "upper"
    );
```

and are used only below

```
public static Schema.Type getScriptFunctionReturnType(String functionName) {
        functionName = functionName.toLowerCase();

        if (dateFunctions.contains(functionName) || stringOperators.contains(functionName)) {
            return Schema.Type.TEXT;
        }

        if (mathConstants.contains(functionName) || numberOperators.contains(functionName)
                || trigFunctions.contains(functionName) || binaryOperators.contains(functionName)
                || utilityFunctions.contains(functionName)) {
            return Schema.Type.DOUBLE;
        }

        throw new UnsupportedOperationException(
                String.format(
                        "The following method is not supported in Schema: %s",
                        functionName));
    }

```


Can we actually get rid of them and use grammar information to get function types from grammar (or vice versa, extract each function definition into some entity that can generate script, and have metadata, and bring it to grammar)?

@dai-chen what do you think?

Reply:
Sure, I agree we should refactor the function in more OO way (which is convenient to integrate into our Logical Plan later). And one thing I want to mention is that the whole type system in new semantic analyzer is based on Elasticsearch data type. But here we are mapping function return type to JDBC type.